packages/squiggle-lang/src/reducer/lambda.ts

```diff
@@ -70,7 +76,13 @@ export class SquiggleLambda extends Lambda {
 
       let localStack = stack;
       for (let i = 0; i < parametersLength; i++) {
-        localStack = localStack.push(parameters[i], args[i]);
+        const parameter = parameters[i];
+        localStack = localStack.push(parameter.name, args[i]);
+        if (parameter.domain && !parameter.domain.value.includes(args[i])) {
+          throw new REOther(
```

Comment:
Can you make an error type for Domain errors?

Reply:
Added. But I'm really unsure about the value of different error types, we don't use them anywhere and it's weird to add ~8 line class for each error.